text/todo-generic-payload.md

```diff
@@ -0,0 +1,368 @@
+
+| Value | Name                                     | Description                                                                |
+| ----- | ---------------------------------------- | -------------------------------------------------------------------------- |
+| 0     | GENERIC_PAYLOAD_FUNCTION_TYPE_LOGICAL    | Logical state control (true/false, on/off, high/low).                      |
+| 1     | GENERIC_PAYLOAD_FUNCTION_TYPE_CONTINUOUS | Continuous value control (e.g., servo angle, motor speed). Uses min/max.   |
+| 2     | GENERIC_PAYLOAD_FUNCTION_TYPE_DISCRETE   | Discrete value selection (e.g., mode number, PWM frequency). Uses min/max. |
+
+#### GENERIC_PAYLOAD_ERROR_FLAGS
+
+Error flags for a generic payload. This is a bitmask enumeration.
+
+| Value | Name                                       | Description                                                                                                              |
+| ----- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------ |
+| 1     | GENERIC_PAYLOAD_ERROR_FLAGS_SOFTWARE_ERROR | There is an error with the generic payload's software.                                                                   |
+| 2     | GENERIC_PAYLOAD_ERROR_FLAGS_HARDWARE_ERROR | There is an error with the generic payload's hardware.                                                                   |
+| 4     | GENERIC_PAYLOAD_ERROR_FLAGS_OVERTEMP_FAULT | A specified limit has been reached by the generic payload's temperature sensor. Applies if payload monitors temperature. |
+| 8     | GENERIC_PAYLOAD_ERROR_FLAGS_CUSTOM         | Generic payload custom failure. See custom error flag bitmask for details.                                               |
+
+### Messages
+
+#### GENERIC_PAYLOAD_STATUS (Message ID: 60000)
+
+Status and discovery message for a generic payload. This should be broadcast at a low rate (nominally 1 Hz) to announce the payload's presence and capabilities.
+
+| Field Name             | Type     | Description                                                          |
+| ---------------------- | -------- | -------------------------------------------------------------------- |
+| uptime_ms              | uint32_t | Time since payload boot (in milliseconds).                           |
+| error_flags            | uint32_t | Current error flags (bitmask). See GENERIC_PAYLOAD_ERROR_FLAGS.      |
+| custom_error_flags     | uint32_t | Bitmap for custom error flags specific to this payload type.         |
+| num_functions          | uint16_t | Number of functions this payload provides.                           |
+| num_telemetry_channels | uint16_t | Number of telemetry channels this payload provides.                  |
+| name                   | char[32] | Name of the generic payload for UI display (NULL-terminated string). |
+
+**Extensions:**
```

Comment:
What do you mean by extensions? Separate messages or mavlink 2 extensions?

Reply:
Either way, these are interesting things to consider.

Reply:
the intent was to be used as mavlink 2 extensions, though that may have been me overthinking it, wanted to minimize the size of the message if they weren't used.

Reply:
I think these look like good things to expose, particularly for removable payloads. They might very reasonable also be useful for cameras and all the other payloads we have. In particular if they allow the vehicle to know that it perhaps has to do some retuning or thinking about how it has to fly.

I suspect though there will be many other things people want. TBD whether they should be extensions. My initial reaction is not. The mass and torque arm would be in the static payload message.